CONTRIBUTING.md

```diff
@@ -0,0 +1,43 @@
+# Welcome
+
+You have stumbled upon a top secret part of the internet where rocket scientists hang out.
+
+It would be a multi-national security concern if you joined a [Discord channel](https://discord.gg/sWdSk6m) where the rocket scientists answer any questions thrown their way.
+
+Of course we are kidding, Space Cloud welcomes you!
+
+## How to Contribute to Space Cloud? (General Guidelines)
+
+There are multiple ways to contribute to Space Cloud:
+- Code contributions.
+- Creating well-described issues with reproducible steps.
+- Community contributions (making examples, tutorials, or spreading the word).
+
+- Looking for a starting point? Search for issues with the label [good first issue](https://github.com/spaceuptech/space-cloud/labels/good%20first%20issue).
+- Every PR should be attached to an issue.
+- If your contribution doesn't align with an existing issue, create a new issue and link it with your PR.
+- Space Cloud's [motivation](https://docs.spaceuptech.com/introduction/motivation/) and [design goals](https://docs.spaceuptech.com/introduction/design-goals/) are the driving factors for feature ideas.
+
+- If you are confused about where to start, which issue to pick or how to solve one, we welcome your questions in the #contributions channel on Discord.
+
+## Sign the CLA
+### Before your PRs can be merged, you need to sign the [Contributor License Agreement](https://cla-assistant.io/spaceuptech/space-cloud).
+
+## Project Structure
+
+- The core components of Space Cloud are Gateway, Runner, and Space CLI.
+- The codebase is a monorepo consisting of the core components and their sub modules.
+
+## Branching Strategy and Guidelines
+
+- Space Cloud follows a version and release based branching practice.
```

Comment:
```suggestion
- Space Cloud follows a version and release based branching practice. Active development happens on the version branches starting with the prefix `v`.
```

Reply:
Also, It makes sense to maintain a CHANGELOG that lists out all changes under each release. It will make it easier for users to look at what changed between versions and there are a lot many benefits to maintaining a CHANGELOG. We could also have a section on versions that are under active development and bugfixes